### PR TITLE
Add warning to mailer documentation about authentication

### PR DESCRIPTION
**Pull request by zeripath**
References #7966

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by mrsdizzie** [comment]
Hmm weird, I use SMTP fine with user/password without setting ```IS_TLS_ENABLED=true``` or localhost set.

Looking at the way Gitea works is that if you set ```IS_TLS_ENABLED=true``` or have port 465 then it tries to starts a TLS connection directly:

https://github.com/go-gitea/gitea/blob/5789e603161aff311f8afcc15585892e783c931e/services/mailer/mailer.go#L160-L164

but if not it still tries StartTLS if supported:

https://github.com/go-gitea/gitea/blob/5789e603161aff311f8afcc15585892e783c931e/services/mailer/mailer.go#L185-L191

Which will upgrade the connection to a TLS connection in most cases. So ```IS_TLS_ENABLED=true``` is somewhat confusing as you will often end up with a TLS connection anyway since StartTLS will work in many cases -- and since it just works you'll never think to change the default IS_TLS_ENABLED setting because you won't have a TLS problem. In the case of the linked issue it doesn't work because that server happens to not support TLS at all.

So I think it should say you can only use username/password if the SMTP server supports TLS.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -398,6 +398,7 @@ set name for unique queues. Individual queues will default to
    the "Name" \<email@example.com\> format.
 - `USER`: **\<empty\>**: Username of mailing user (usually the sender's e-mail address).
 - `PASSWD`: **\<empty\>**: Password of mailing user.  Use \`your password\` for quoting if you use special characters in the password.
+   - NB: Go's internal libraries will only permit authentication if `IS_TLS_ENABLED=true` or `HOST=localhost`.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
   - Please note: authentication is only supported when `IS_TLS_ENABLED=true` or `HOST=localhost` due to limitations imposed by the Go internal libraries.
```
I had to look for _Nota Bene_! 😁

File: docs/content/doc/usage/email-setup.en-us.md [hunk 2]
```diff
@@ -46,6 +46,8 @@ PASSWD         = `password`
 
 For the full list of options check the [Config Cheat Sheet]({{< relref "doc/advanced/config-cheat-sheet.en-us.md" >}})
 
+- NB: Go's internal libraries only permit authentication on localhost or `IS_TLS_ENABLED=true`.
```
Comment (guillep2k) [inline docs/content/doc/usage/email-setup.en-us.md]:
```suggestion
- Please note: authentication is only supported when `IS_TLS_ENABLED=true` or `HOST=localhost` due to limitations imposed by the Go internal libraries.
```

**Comment by zeripath** [comment]
@mrsdizzie - I forgot about that little feature - fancy improving @guillep2k's suggestion?

@guillep2k presumably nota bene has died out in the Americas, I thought it was well-known e.g. i.e. etc.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md (hunk 1 above)
Comment (zeripath) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
   - Please note: authentication is only supported when the SMTP server supports TLS through StartTLS, `IS_TLS_ENABLED=true` or is `HOST=localhost` due to protections imposed by the Go internal libraries.
```

File: docs/content/doc/usage/email-setup.en-us.md (hunk 2 above)
Comment (zeripath) [reply docs/content/doc/usage/email-setup.en-us.md]:
```suggestion
   - Please note: authentication is only supported when the SMTP server supports TLS through StartTLS, `IS_TLS_ENABLED=true` or is `HOST=localhost` due to protections imposed by the Go internal libraries.
```

**Comment by silverwind** [comment]
Can you also add this to `app.ini.sample`?

**Comment by mrsdizzie** [comment]
@zeripath looks good. Maybe while in here we can update the comment for```IS_TLS_ENABLED=true``` that it actually means to just try Implicit TLS first. I can't make a code suggestion since it isn't part of the diff but maybe:

```
Try an Implicit TLS connection first. WIll fall back to STARTTLS if not supported. Implicit TLS is also implied when using port 465 in the `HOST` value.
```

Since the name is confusing and the current comment isn't really descriptive of what it does (most people read it as required for any TLS and it is probably wrongly suggested for servers or ports that don't support Implicit TLS).

**Comment by zeripath** [comment]
@mrsdizzie do you mean explicit rather than implicit here? My understanding is that STARTTLS is opportunistic TLS or implicit - in that the server and the client will upgrade to TLS implicitly without us explicitly telling them to, similarly connecting to a TLS port and detecting it was a TLS connection would be implicit,  but saying connect with TLS is being explicit.

**Comment by zeripath** [comment]
Oh this is a weirdness of the RFC - the implicit here means that at the protocol layer the TLS is implicit. (even though the user has made it very explicit that they want TLS.)

**Comment by mrsdizzie** [comment]
@zeripath yes agree its all very confusing : (

Most simply, STARTTLS is considered the explicit one because the connection is explicitly upgraded if not secure. Other option is implicit because its implied the connection is already secure.

**Comment by zeripath** [comment]
OK I've tried to make that a bit clearer.

Fundamentally `IS_TLS_ENABLED` needs to be changed - it's confusing and should be `FORCE_SMTPS_CONNECTION`.

**Review by mrsdizzie — APPROVED** [review]
lgtm -- hopefully this will be just as helpful to us in the future as a reminder of how it works : )

**Review by guillep2k — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**